src/InventoryManager.cpp

```diff
@@ -78,3 +79,68 @@ int InventoryManager::userInput()
     
 	return 0;
 }
+
+/*uses ActiveInventory functions to create items from a csv file*/
+void InventoryManager::readCSVFile(const std::string &file)
+{
+	std::string name, str_id, cat, sub_cat, qty, sale_price;
+	std::string tax, total_price, buy_cost, profit, exp, tmp_line; 
+	unsigned long id;
+	
+	std::ifstream csv_file(file);
+
+	if(!csv_file.is_open()) std::cout << "ERROR: File Open" << '\n';
+	
+	/* This skips the first line for you. */
+	std::getline(csv_file, tmp_line);
+
+	while(csv_file.good()){
+
+		/*reading in the csv info, converting types*/
+		getline(csv_file, name, ',');
+
+		/*exit the loop if we domaken't read in a new name.*/
+		if (!csv_file.good()) break;
+
+		getline(csv_file, str_id, ',');
+		id = std::stoul(str_id);
+		getline(csv_file, cat, ',');
+		getline(csv_file, sub_cat, ',');
+		getline(csv_file, qty, ',');
+		getline(csv_file, sale_price, ',');
+		getline(csv_file, tax, ',');
+		getline(csv_file, total_price, ',');
+		getline(csv_file, buy_cost, ',');
+		getline(csv_file, profit, ',');
+		getline(csv_file, exp, '\n');
+
+		/* Add the item to the inventory. */
+		if (active_inventory->addItem(name, cat, id) != -1) {
+			std::cout << "Added " << name << " of type " << cat << std::endl;
+		} else {
+			fprintf(stderr, "Failed to read in item %s with ID %lu.\n", name.c_str(), id);
+		}
+
+		/* Update all the item categories */
+		active_inventory->updateItem(name, "sub_category", sub_cat);
+		active_inventory->updateItem(name, "quantity", qty);
+		active_inventory->updateItem(name, "sale_price", sale_price);
+		active_inventory->updateItem(name, "buy_cost", buy_cost);
+		active_inventory->updateItem(name, "tax", tax);
+		active_inventory->updateItem(name, "total_price", total_price);
+		active_inventory->updateItem(name, "profit", profit);
+		if (cat == "Perishable") {
+			active_inventory->updateItem(name, "expiration_date", exp);
+		}
+
+		/* Test to see if we successfully read in item. */
+		auto item_ptr = active_inventory->searchByName(name);
```

Comment:
You can probably take this out since we don't care if the item_ptr is not null, that means it exists.
I'd change this to:
```
auto item_ptr = active_inventory->searchByName(name);

if (item_ptr == NULL) {
    fprintf(stderr, "Failed to read item %s that we just created.\n", name.c_str());
}
```